state/execution.go

```diff
@@ -50,6 +50,18 @@ func BlockExecutorWithMetrics(metrics *Metrics) BlockExecutorOption {
 	}
 }
 
+// BlockCommitExecParams provides context for executing the block commit.
+type BlockCommitExecParams struct {
```

Comment:
I'm a little worried that we're making a struct of a lot of things that are pointers. I don't know if there's anything better to be done, but... uff.

Reply:
removed the pointer of the validator.
for the block pointer, it requires bigger changes, should I do it in this PR?